### ISSUE TITLE
wrong line in error message

### ISSUE THREAD

**Issue by rodneyrehm** [sass/libsass#972]
Considering the following (invalid) input:

```scss
$foo: 123px;

.bar {
  width: $bar;
}

bad-token-test
```
I get the error reported on *line 5* (`3.2.1-beta.2`) and the formatted error message looks as follows:

```
Error: invalid top-level expression
        on line 5 of stdin
>> }
   -^
```

`3.1.0` reports the error on *line 7*

**Comment by xzyfer** [comment]
@mgreter any ideas here?

**Comment by mgreter** [comment]
Whitespace was not "actively" lexed, so position still points to the position it is reporting. Maybe we should move the parser when an error is thrown (we attach the prefixed whitespace to each lexed token, so we should _not_ just munch away whitespace on it own).

**Comment by mgreter** [comment]
OK, the actual bug is caused by the way `lex` works together with the `error` function in `parser.cpp`. Normaly when we report errors, it is about the thing we actually lexed. But with this error we actually already lexed the whitespace (and that's why the error "somehow correctly"  points there). One easy fix is to set `before_token` to `after_token` before calling the error (both are members of the Parser class and contain line/col positions).

**Comment by rodneyrehm** [comment]
Are you sure this is fixed? I'm still getting the wrong line reported…

**Comment by mgreter** [comment]
I was pretty sure it was fixed, but it looks your correct => #1002

**Comment by mgreter** [comment]
Added a new commit which will hopefully solve it now (https://github.com/sass/libsass/commit/8d8dc08fdff06e33ccde3725003837968d3f1ae0)!
